akd/tools/reranker.py

```diff
@@ -153,9 +156,428 @@ def __str__(self) -> str:
         return self.__class__.__name__
 
 
+# LLM-based Reranker Components
+
+
+class ScoringCategory(BaseModel):
+    """Scoring category with name, description, and numeric value."""
+
+    name: str = Field(..., description="Category name (e.g., 'Perfect Match', 'Acceptable', 'Unusable')")
+    description: str = Field(..., description="Detailed description of what this category represents")
+    value: float = Field(..., description="Numeric score for this category")
+
+
+class ScoringCriterion(BaseModel):
+    """Individual criterion for evaluating results."""
+
+    name: str = Field(..., description="Criterion name (e.g., 'Relevancy', 'Processing Level', 'Ease of Use')")
+    description: str = Field(..., description="Detailed description of what this criterion evaluates")
+    weight: float = Field(default=1.0, ge=0.0, le=1.0, description="Weight for this criterion (0.0 to 1.0)")
+    scoring_categories: list[ScoringCategory] = Field(
+        default_factory=lambda: [
+            ScoringCategory(
+                name="Perfect Match",
+                description="Result fully satisfies the criterion requirements",
+                value=3.0,
+            ),
+            ScoringCategory(
+                name="Acceptable",
+                description="Result partially satisfies the criterion requirements",
+                value=2.0,
+            ),
+            ScoringCategory(
+                name="Unusable",
+                description="Result does not satisfy the criterion requirements",
+                value=0.0,
+            ),
+        ],
+        description="Scoring categories specific to this criterion with their numeric values",
+    )
+
+
+class CriterionScore(BaseModel):
+    """Score for a single criterion."""
+
+    category: str = Field(..., description="The selected category (e.g., 'Perfect Match', 'Acceptable', 'Unusable')")
+    reasoning: str = Field(..., description="Brief explanation for the score")
+
+
+class LLMRerankerToolConfig(RerankerToolConfig):
+    """Configuration for LLM-based reranker."""
+
+    base_url: AnyUrl | None = Field(default=None, description="Base URL for LLM API")
+    api_key: str | None = Field(default=None, description="API key for LLM")
+    model_name: str = Field(default="gpt-4o-mini", description="LLM model name")
+    temperature: float = Field(default=0.0, ge=0.0, le=2.0, description="LLM temperature")
+    agent_system_prompt: str = Field(
+        default=(
+            "You are an expert at evaluating search results. "
+            "Analyze the provided result for the query against all given criteria and "
+            "select the most appropriate category for each. Provide clear reasoning."
+        ),
+        description="System prompt for the internal scoring agent",
+    )
```

Comment:
have this get from os.getenv as well so that we can globally modify whenever we want...and default to this value you have. something like `AKD_LLM_RERANKER_SYSTEM_PROMPT`